packages/dds/merge-tree/src/test/client.rollback.spec.ts

```diff
@@ -74,6 +74,132 @@ describe("client.rollback", () => {
+        client.insertMarkerLocal(
+            0,
+            ReferenceType.Simple,
+            {
+                [reservedMarkerIdKey]: "markerId",
+                foo: "bar",
+            },
+        );
+        const marker = client.getMarkerFromId("markerId") as Marker;
+        client.annotateMarker(marker, { foo: null }, undefined);
+        client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+
+        const properties = marker.getProperties();
+        assert.equal(properties?.foo, "bar");
+    });
+    it("Should rollback annotate marker rewrite", async () => {
+        client.insertMarkerLocal(
+            0,
+            ReferenceType.Simple,
+            {
+                [reservedMarkerIdKey]: "markerId",
+                foo: "bar",
+            },
+        );
+        const marker = client.getMarkerFromId("markerId") as Marker;
+        client.annotateMarker(marker, { abc: "def" }, { name: "rewrite" });
+        client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+
+        const properties = marker.getProperties();
+        assert.equal(properties?.foo, "bar");
+        assert.equal(properties?.abc, undefined);
+    });
+    it("Should rollback annotate causes split string", async () => {
+        client.insertTextLocal(0, "abcdefg");
+        client.annotateRangeLocal(1, 3, { foo: "bar" }, undefined);
+        client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+
+        for (let i = 0; i < 4; i++) {
+            const props = client.getPropertiesAtPosition(i);
+            assert(props === undefined || props.foo === undefined);
+        }
+    });
+    it("Should rollback annotate over split string", async () => {
+        client.insertTextLocal(0, "abfg");
+        client.insertTextLocal(1, "cde");
+        client.annotateRangeLocal(1, 6, { foo: "bar" }, undefined);
+        client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+
+        for (let i = 0; i < 7; i++) {
+            const props = client.getPropertiesAtPosition(i);
+            assert(props === undefined || props.foo === undefined);
+        }
+    });
+    it("Should rollback annotate with same prop", async () => {
+        client.insertTextLocal(0, "abcde");
+        client.annotateRangeLocal(2, 3, { foo: "bar" }, undefined);
+        client.annotateRangeLocal(1, 4, { foo: "bar" }, undefined);
+        client.rollback?.({ type: MergeTreeDeltaType.ANNOTATE }, client.peekPendingSegmentGroups());
+
+        for (let i = 0; i < 5; i++) {
+            const props = client.getPropertiesAtPosition(i);
+            if (i === 2) {
+                assert.equal(props?.foo, "bar");
+            } else {
+                assert(props === undefined || props.foo === undefined);
+            }
+        }
+    });
+    it("Should zamboni rolled back annotated segment", async () => {
```

Comment:
This test fails. Not sure why the segment created from annotating doesn't get zamboni'ed. Should it?